glusterd2/xlator/load.go

```diff
@@ -127,17 +129,14 @@ func loadXlator(xlPath string) (*Xlator, error) {
 
 func getXlatorsDir() string {
 
-	// glusterfs gets the path to xlator dir from a compile time flag named
-	// 'XLATORDIR' which gets passed through a -D flag to GCC. This isn't
-	// available to external programs via gluster CLI yet. When one or more
-	// versions of gluster are installed from source or otherwise, the
-	// following is the most fool-proof but hacky way to get the xlator dir
-	// location without making assumptions.
-
-	cmd := "strings -d `which glusterfsd` | awk '/glusterfs\\/.*\\/xlator$/'"
-	out, err := exec.Command("sh", "-c", cmd).Output()
+	out, err := exec.Command("gluster", "--print-xlatordir").Output()
```

Comment:
this brings in a hard dependency of shipping gluster cli always with GD2. I don't think its a good approach.

Reply:
Good catch!

Reply:
Looks fine till this(https://github.com/gluster/glusterd2/issues/534) gets implemented

Reply:
Good catch @atinmu but for atleast another 3 months, for users using/testing the feature, we should have an option to not crash. Should be very useful IMO.

Reply:
I'm OK to accept this patch as long as there's an issue filed to correct this later.

Reply:
#534 captures these requirements.